containers/unit_tests/TestDynRankView_TeamScratch.hpp

```diff
@@ -0,0 +1,70 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <Kokkos_DynRankView.hpp>
+
+namespace Test {
+
+void test_dyn_rank_view_team_scratch() {
+  using execution_space = TEST_EXECSPACE;
+  using memspace_type   = execution_space::scratch_memory_space;
+  using drv_type        = Kokkos::DynRankView<int, memspace_type>;
+  using policy_type     = Kokkos::TeamPolicy<execution_space>;
+  using team_type       = policy_type::member_type;
```

Comment:
Nitpick: Defining a whole lot of aliases that are only used once when really only the execution space is repeated.
I would drop them and template the whole thing on ExecutionSpace.
Not blocking obviously.